Add troubleshooting JH doc (#32)
aicoe-aiops/data-science-workflows: docs/troubleshooting_jh.md

> # Troubleshooting JupyterHub for Data Scientists

Reviewer: Overall Comment: It might be good idea to split this doc into a couple of sections for example: 1. a list of specific known issues and solutions for how to fix them, like "kernel is crashing, must restart pod with more resources". 2. A more general section on the available monitoring techniques for RAM, PVC, CPU, and Logs that are available letting folks know where they can go to diagnose different issues. WYDT?

Author: That sounds good, although I intended the doc to only talk about the second section since @hemajv mentioned that there are [runbooks](https://github.com/operate-first/blueprint/tree/main/docs/userstories/jupyterhub) for known issues. I still think there's value in adding known issues with screenshots to the user document. If that's what is reasonable, we should update the original [google doc](https://docs.google.com/document/d/1Y83HEUQbwilc1K4gpVkt79VXH3T02TSzTFhPwvMVXSY/edit)  and then with all the additional suggestions we can change this PR. I'll take most suggestions from here and work on it but feel free to add more known issues, and content.

> Data scientists using Jupyterhub may encounter problems while running their workloads. This document highlights how to troubleshoot some common problems. If you get an error that is not explained in this document, make sure to create an issue in the [support repository](https://github.com/operate-first/support) for the Operate First project. 

Reviewer: ```suggestion
Data scientists using Jupyterhub may encounter problems while running their notebooks. This document highlights how to troubleshoot some common problems. If you get an error that is not explained in this document, make sure to create an issue in the [support repository](https://github.com/operate-first/support) for the Operate First project. 
```

# Resource usage

> ## RAM

Reviewer: Can you show a screen shot of reproducing this error?

A lot of the common problems are related to the memory consumption of the workload. If your notebook requires more RAM than what is requested by your instance, you will likely see your kernel crash. In this case, you can look at your memory consumption chart:

*   If you are using an Elyra image, you can see a sum of memory used by all your active kernels at the bottom of your screen.

![Elyra notebook RAM](./../public/assets/troubleshooting_jh/elyra_memory.png)

*   You could also go to the [Grafana dashboard](https://grafana-route-opf-monitoring.apps.zero.massopen.cloud/d/YuYfkHYMk/jupyterhub-usage?orgId=1&var-user_id=shanand-40redhat-2ecom) and select your username to see the resource usage for your pod.

> *   Currently, you can spawn a pod with memory ranging from 1GB to 32GB.

Reviewer: ```suggestion
*   If your jupyterhub pod continues to crash due to insufficient memory, you should shut down your jupyterhub server and spawn a new one with more resources. The current default instances go up to 32GB of RAM. If more is needed additional resources can be requested from [operate first support](https://github.com/operate-first/support) 
```

## PVC

> You could also run into problems spawning your pods and this could be because your PVC is full. If you are not able to spawn your pod, check the PVC status in the [grafana dashboard ](https://grafana-route-opf-monitoring.apps.zero.massopen.cloud/d/YuYfkHYMk/jupyterhub-usage)and if it is full, request for more space on the support repository for Operate First. You  can also find this information in your Jupyterhub terminal by using the following command:

Reviewer: Can you add a screen shot of this error as well?

> You could also run into problems spawning your pods and this could be because your PVC is full. If you are not able to spawn your pod, check the PVC status in the [grafana dashboard ](https://grafana-route-opf-monitoring.apps.zero.massopen.cloud/d/YuYfkHYMk/jupyterhub-usage)and if it is full, request for more space on the support repository for Operate First. You  can also find this information in your Jupyterhub terminal by using the following command:

Reviewer: We've also seen that we can't save any files in a spawned pod when the pvc gets filled while you are running it.

> df -h /opt/app-root/src 

Reviewer 2: This looks good. Just a nit: Can you make it 
`df -h /opt/app-root/src`
instead.

Reviewer: yes, please add the back-ticks to format the `df` command.

## CPU

> To monitor the real time CPU and GPU resources used by your data science workload, check the [grafana dashboard](https://grafana-route-opf-monitoring.apps.zero.massopen.cloud/d/YuYfkHYMk/jupyterhub-usage) for spikes and falls. 

Reviewer: Is there an example of a CPU failure?

> If you can not understand the issues that you are experiencing, you could check the [pod logs](https://console-openshift-console.apps.zero.massopen.cloud/k8s/ns/opf-jupyterhub/pods) for your instance. If you do not have access to the logs, reach out to the support repository or the slack channel.

Reviewer: Can you add a link to all mentions of "support repository" or "slack channel"

> 

Reviewer 3: ```suggestion
# Common Issues

## No Kernel Error

If you see a "No Kernel" error at any point in time, such as when there are no Kernel options available to select from on the top right side of your JupyterHub window

You can reinstall the Kernel by running the following command
`ipython kernel install --user --name=python3 --display-name 'Python 3'`

```
![image](https://user-images.githubusercontent.com/32435206/114927132-281ad300-9dff-11eb-8f37-cfb7b3ef2971.png)

Can you also add a small note on the No Kernel issue. I been experiencing this even today while switching between JH images.
